### DISCUSSION TITLE
How to input external data in the Pybroker backtest

### DISCUSSION THREAD

**Discussion by victorsmoreschi** [edtechre/pybroker#147] [Q&A] [answered]
I am working on a project where I try to project the future price of an asset. To do this, my intention is to train the model with a serie of external data that I collected (some CDS, country premiuns, etc). How could I input this inside the code of training suggested in this link (https://www.pybroker.com/en/latest/notebooks/6.%20Training%20a%20Model.html)? After that, run the backtest for the model. Thanks for your help!

**Comment by edtechre** [comment, marked as answer]
Hi @victorsmoreschi,

The easiest way to do this is to probably load your data into a Pandas DataFrame, and then use the DataFrame as a custom Data source. See this notebook: https://www.pybroker.com/en/latest/notebooks/7.%20Creating%20a%20Custom%20Data%20Source.html#Using-a-Pandas-DataFrame
